project-fortis-pipeline/ops/storage-ddls/cassandra-setup.cql

```diff
@@ -120,83 +127,69 @@ CREATE TABLE fortis.events(
     PRIMARY KEY (eventid)
 );
 
-// *****************************************************************************
-// * Views
-// *****************************************************************************
+// topics 2 and 3 are always written as empty string. They're included to keep partition keys the same as related aggregations.
+CREATE TABLE fortis.populartopics(
+    pipelinekey text,
+    conjunctiontopic1 text,
+    conjunctiontopic2 text,
+    conjunctiontopic3 text,
+    tilez int,
+    tileid text,
+    externalsourceid text,
+    periodtype text,
+    perioddate timestamp,
+    mentioncount counter,
+    avgsentimentnumerator counter,
+    PRIMARY KEY ((pipelinekey, conjunctiontopic1, conjunctiontopic2, conjunctiontopic3, tilez, tileid, externalsourceid, periodtype), perioddate)
+) WITH CLUSTERING ORDER BY (perioddate DESC);
+
+CREATE TABLE fortis.eventplacesbysource(
+    pipelinekey text,
+    conjunctiontopic1 text,
+    conjunctiontopic2 text,
+    conjunctiontopic3 text,
+    tilez int,
+    tileid text,
+    externalsourceid text,
+    periodtype text, // always set to day
+    eventtime timestamp,
+    eventid text,
+    placeid text,
+    centroidlat double,
+    centroidlon double,
+    insertiontime timestamp,
+    PRIMARY KEY ((pipelinekey, conjunctiontopic1, conjunctiontopic2, conjunctiontopic3, tilez, tileid, externalsourceid, periodtype), eventtime, eventid, placeid)
+) WITH CLUSTERING ORDER BY (eventtime DESC);
+
+CREATE TABLE fortis.popularsources (
+    pipelinekey text,
+    conjunctiontopic1 text,
+    conjunctiontopic2 text,
+    conjunctiontopic3 text,
+    tilez int,
+    tileid text,
+    periodtype text,
+    perioddate timestamp,
+    externalsourceid text,
+    mentioncount counter,
+    avgsentimentnumerator counter,
+    PRIMARY KEY ((pipelinekey, conjunctiontopic1, conjunctiontopic2, conjunctiontopic3, tilez, tileid, periodtype), perioddate, externalsourceid)
+) WITH CLUSTERING ORDER BY (perioddate DESC);
+
+CREATE TABLE fortis.eventsbypipeline(
+    pipelinekey text,
+    conjunctiontopic1 text,
+    conjunctiontopic2 text,
+    conjunctiontopic3 text,
+    tilez int,
+    tileid text,
+    externalsourceid text, // TODO: should this be part of the Primary Key?
```

Comment:
This table is used to drive the facts page which displays a list of all the events that happened, broken down by pipeline and filterable by keyword and date. As such, I don't think the externalsourceid is required as part of the key: we essentially just need a way to get the eventids for a given set of keywords, pipeline and date.

Reply:
Makes sense. My concern was that it'd be possible for rows with the same primary key to collide even if they had different `externalsourceid`s, but I suppose we'd never see a case where two inserts could have the same `eventid` but different `externalsourceid`.